Client/src/Unified/Services/Submitter/BlockRequest.cs

```diff
@@ -0,0 +1,51 @@
+// This file is part of the ArmoniK project
+//
+// Copyright (C) ANEO, 2021-$CURRENT_YEAR$. All rights reserved.
+//   W. Kirschenmann   <wkirschenmann@aneo.fr>
+//   J. Gurhem         <jgurhem@aneo.fr>
+//   D. Dubuc          <ddubuc@aneo.fr>
+//   L. Ziane Khodja   <lzianekhodja@aneo.fr>
+//   F. Lemaitre       <flemaitre@aneo.fr>
+//   S. Djebbar        <sdjebbar@aneo.fr>
+//   J. Fonseca        <jfonseca@aneo.fr>
+//   D. Brasseur       <dbrasseur@aneo.fr>
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY, without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Threading;
+
+using ArmoniK.DevelopmentKit.Client.Common;
+using ArmoniK.DevelopmentKit.Common;
+
+using JetBrains.Annotations;
+
+namespace ArmoniK.DevelopmentKit.Client.Unified.Services.Submitter;
+
+internal class BlockRequest : IDisposable
+{
+  public IServiceInvocationHandler Handler;
+
+  [CanBeNull]
+  public ArmonikPayload Payload { get; set; }
+
+  public SemaphoreSlim Lock { get; set; }
+
+  public void Dispose()
+  {
+    Payload?.Dispose();
+    Payload = null;
+    Lock?.Release();
```

Comment:
Why do you release the semaphore in dispose?

Reply:
It can be discussed indeed. The blockRequest shall have to be released at the end of dataflow. The blockrequest must be released and this is the safest place to signal that another blockRequest can be created by a new SubmitAsync. Don't you think?

Reply:
Removed Release in dispose and called before in Actionblock Service.cs